### PR TITLE
refactor: signin 

### PR DESCRIPTION
**Pull request by xudaotutou**
### Summary
:sparkles::lock::recycle:

This pull request adds new features and improvements to the frontend desktop application, such as SMS messaging, GitHub and WeChat OAuth, and more apps. It also updates the dependencies, the layout component, the API handlers, and the authentication logic. It also removes or ignores some files that are not needed or secure for production.

> _Oh we're the coders of the sea, we build the apps with `ts`_
> _We use the `docker` and the `kube`, to make our code run smooth_
> _We add the `auth` and the `sms`, to sign in users with ease_
> _And when we're done we push and pull, and sing this shanty tune_

### Walkthrough
*  Add and update dependencies for new features ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-e096f3d2744612bf7e830137543c51d72c59489d165d7d230b8101bb65a84d85L12-R23), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-e096f3d2744612bf7e830137543c51d72c59489d165d7d230b8101bb65a84d85L28-R36), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-e096f3d2744612bf7e830137543c51d72c59489d165d7d230b8101bb65a84d85L40-R59))
*  Add and modify API handlers for authentication with phone, SMS, GitHub, and WeChat ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-925606db31dc658a56f861280efa2cc9605b0596cbce6f3feaddd1154c2ee64fR1-R73), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-21368d22d1932ae6d7699802de1c301999408c7a626a9f533f3c010bb5d4aa11L1-R76), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-c86fe249abaeecd88a600540e504085af975ff9f06011fea86b5776e605922bdL1-R61), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-a8e2c2a1a3cd2096c41f7c30bacb08012a7d637df8bb47593b660eccd3a0d3a0L1-R63))
*  Add and modify page components for sign in and OAuth callbacks ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-beec8ccd426771344314475a99fc825b9e3ffea2c8e8abb5a2256f94236770b2L1-R34), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-f4daef00368888e45ceb00f6832c7e2fec248cf288a93475e443cee60545332cR1-R316), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-4ef0e9fe4def0cba10300c5990451ec3998600519d2f11d8c25e4f5450ffee77L1-R35), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-87d92c16e255847ac28c8a9138da4f0354e93228917e0e6bb94cf3181b191715))
*  Add and modify helper functions for generating and verifying JSON web tokens ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-af34e0df3a3d9dfbbb81a909e70c79b74a3ccfa553a9484b9917d77d2403f67cL3-R39))
*  Move and update context provider and components for more apps feature ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-4ae759586b8ce6e86c0d3949c7e588f4dcfd84aa3232cf567704c324c85196abL2-R5), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-4ae759586b8ce6e86c0d3949c7e588f4dcfd84aa3232cf567704c324c85196abL31-R16), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-942cf93ff546b74643d32be5bdb582b156157a33ace9bb5bf07ca080c1336c60L3-R3), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-322ecb0d4c3fbb372838a948e5d24b716b5ec69c8b50fcf308b74fba2f468bb3L1-R1), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-c04141c0579122cad0175ca367c6a04f583111f500a9211b58fa3b207d2b2b53L3-R42))
*  Remove and update imports and logs for desktop content feature ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-4ae759586b8ce6e86c0d3949c7e588f4dcfd84aa3232cf567704c324c85196abL2-R5), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-3d93354970e557abc09c23eefead6604d11f2c82a98187b81979b7f6746dade0L1), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-3d93354970e557abc09c23eefead6604d11f2c82a98187b81979b7f6746dade0R12), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-3d93354970e557abc09c23eefead6604d11f2c82a98187b81979b7f6746dade0L30), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-3d93354970e557abc09c23eefead6604d11f2c82a98187b81979b7f6746dade0L47))
*  Add and comment out environment variables for development and production ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-d924015f5831c09faf54cb665059d423e79295e0691ec289d300b59db19c25ddL4-R4), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-b08d4e5eff4eec55e0b787348ad6655e4f50a728f6b71a83914598ddd06b16abR1-R19))
*  Add and update files to ignore for docker and git ([link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-1352d2bd12a1f6b933d9bcf7c0e46fe81e2ab3dcc70eec7a50719b3db7f53ddfR8-R9), [link](https://github.com/labring/sealos/pull/3247/files?diff=unified&w=0#diff-981f561d68571d22f5e2cb10b3864e929c459ec47598d82723fd565c37dc5c7fR41-R42))

### CODE REVIEW

File: frontend/desktop/src/services/backend/kubernetes.ts [hunk 1]
```diff
@@ -35,10 +39,113 @@ export function K8sApi(config: string): k8s.KubeConfig {
       }
     });
   }
-
   return kc;
 }
 
+export async function getUser(uid: string) {
```
Comment (zzjin) [inline frontend/desktop/src/services/backend/kubernetes.ts:45]: Move detail function to spec ts files, do not put logic functions inside common base func file.

File: frontend/desktop/src/pages/githubCallback.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  useEffect(() => {
+    if (!router.isReady) return;
+
+    const { code } = router.query;
+    if (code === undefined || code === '') return;
+
+    (async () => {
+      try {
+        const data = await request.post<any, { data: Session, code: number }>('/api/auth/oauth/github', { code })
+        console.log(data)
```
Comment (zzjin) [inline frontend/desktop/src/pages/githubCallback.tsx:20]: remove all test logs when commit

File: frontend/desktop/src/pages/api/desktop/getInstalledApps.ts [hunk 3]
```diff
@@ -10,6 +9,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     const kc = await authSession(req.headers);
 
     const kube_user = kc.getCurrentUser();
+    console.log('getinstallapps')
```
Comment (zzjin) [inline frontend/desktop/src/pages/api/desktop/getInstalledApps.ts:12]: remove all test logs when commit

File: frontend/desktop/src/pages/api/auth/oauth/github/index.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        data: 'access_token is null'
+      })
+    }
+    const userUrl = `https://api.github.com/user`;
+    const { login: name, id, avatar_url } = (await (await fetch(userUrl, {
+      headers: {
+        Authorization: `Bearer ${access_token}`,
+      }
+    })).json()) as TgithubUser;
+    console.log(name, id, avatar_url)
```
Comment (zzjin) [inline frontend/desktop/src/pages/api/auth/oauth/github/index.ts]: Remove all debug `consloe.log`

File: frontend/desktop/src/pages/api/auth/oauth/github/index.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      })
+    }
+    const token = generateJWT(user.id)
+    return jsonRes<Session>(res, {
+      data: {
+        token,
+        user,
+        kubeconfig
+      },
+      message: 'Successfully'
```
Comment (zzjin) [inline frontend/desktop/src/pages/api/auth/oauth/github/index.ts:62]: return code 200?

File: frontend/desktop/src/stores/session.ts [hunk 6]
```diff
@@ -36,9 +44,24 @@ const useSessionStore = create<SessionState>()(
...
+          let result = state === get().oauth_state;
+          set({ oauth_state: undefined });
+          return result
+        },
+        setProvider: (provider?: 'github'|'wechat'|'phone') => {
+          set({ provider });
         }
       })),
-      { name: sessionKey }
+      { name: sessionKey,
```
Comment (zzjin) [inline frontend/desktop/src/stores/session.ts:62]: Pretty print this file

File: frontend/desktop/src/services/backend/db/user.ts [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    case 'default':
+      user = await users.updateOne({ uid: id }, { $set: _user });
+      break;
+    default:
+      throw new Error('provider error')
+      break;
+  }
+  return user
+}
+export async function removeUser({ id, provider, name, avatar_url }: {
```
Comment (zzjin) [inline frontend/desktop/src/services/backend/db/user.ts:139]: Un-unsed, and will never delete user unless user self quest.

File: frontend/desktop/src/services/backend/db/user.ts [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+      user.uid = id
+      await users.insertOne(_user);
+      break;
+    default:
+      throw new Error('provider error')
+      break;
+  }
+  return user
+}
+export async function updateUser({ id, provider, name, avatar_url }: {
```
Comment (zzjin) [inline frontend/desktop/src/services/backend/db/user.ts:106]: Never used? Maybe we need to update username/avatar when user re-login each time.

File: frontend/desktop/src/components/layout/index.tsx [hunk 9]
```diff
@@ -1,41 +1,19 @@
...
-    (async () => {
-      // Initialize, get user information, install application information, etc.
-      await init();
-    })();
-  }, [init]);
 
   return (
     <>
       <Head>
         <title>sealos Cloud</title>
```
Comment (fanux) [inline frontend/desktop/src/components/layout/index.tsx:10]: Sealos Cloud

**Pull request closed without merging**